### PR TITLE
Remove major code smells from the source code

### LINKED ISSUE #8752: Fix RegExp vulnerability and clean up major code smells
### Description

As the title says the issue is about fixing regular expression vulnerability found in the helper that checks if the value is a numeric value and implementing a GHA workflow with Code Quality tool that requires clean up major code smells.

### Your environment
* Handsontable version: -
* Browser Name and version: -
* Operating System: -

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR removes all major code smells from the source code detected by the SonarCloud tool. Additionally, the PR fixes regular expression (marked as "risk") responsible for detecting numeric values. The expression causes performance degradation when an empty string is passed.

**Test code snippet:**
`Handsontable.helper.isNumeric(' '.repeat(100000));`

**Results:**
|                  | PR      | 9.0.2   |
|------------------|---------|---------|
| `isNumeric` helper | ~0.37ms | ~7500ms |

I analyzed this branch after my changes and the results can be found here https://sonarcloud.io/summary/new_code?id=handsontable_handsontable&pullRequest=8742.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. fixes #8752
2.
3.

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+sonar.projectKey=handsontable_handsontable
+sonar.organization=handsontable
+sonar.projectName=Handsontable
+sonar.sources=src,wrappers
```
Comment (jansiegel) [inline sonar-project.properties:4]: @budnix I'm not sure if I understand it correctly - does this include only `src` from `handsontable`, but entire projects (including the config files, etc) for the wrappers? Should it be that way?

Comment (budnix) [reply sonar-project.properties:4]: It means that looking from the root projects only all files within `src` and `wrappers` (except `**/__tests__/**,**/test/**,**/*.spec.js,**/*.spec.ts`) will be analyzed.

Comment (jansiegel) [reply sonar-project.properties:4]: So `./wrappers/react/rollup.config.js` would be analyzed, but `./webpack.config.js` wouldn't?

Comment (budnix) [reply sonar-project.properties:4]: As far as that files do not interfere with the general code quality score I don't mind to analyzed that file as well. I didn't want to create a long exclusion list.

Comment (jansiegel) [reply sonar-project.properties:4]: I guess after moving HOT to a subdirectory, we'll change it to `handsontable, wrappers`.

**Review by jansiegel — APPROVED** [review]

File: sonar-project.properties (hunk 1 above)
Comment (budnix) [reply sonar-project.properties:4]: Right 👌

**Pull request merged**